cosign/README.md

```diff
@@ -0,0 +1,137 @@
+    - Timestamps: Time is a critical component of Sigstore. It’s used to verify that a short-lived certificate issued by Fulcio was valid at a previous point, when the artifact was signed.
+
+## Scenarios
+
+`ratify-verifier-go` supports multiple verification scenarios based on different signing methods, artifact types, and trust sources. These scenarios can be categorized into the following main types.
+
+| Verification Scenario             | Purpose                                                                | Use Case                                                                    |
+|-----------------------------------|------------------------------------------------------------------------|-----------------------------------------------------------------------------|
+| **Keyless Verification**          | Verifies signatures from Rekor, trust material (signed certificate timestamp) from the CT log, and certificates that chain up to Fulcio. | CI/CD pipelines or automated workflows where no private key management is needed. |
+| **Key-Based Verification**        | Verifies signature using a known **public key**.                        | Using an on-disk public key, or a key provided by other providers (e.g., "KMS provider") |
+| **Timestamp Verification**        | Verifies the **timestamp** of the signature to prevent time-based attacks. Not standalone verification. | Sigstore clients relying on Rekor to provide the timestamp use the entry’s inclusion time. Sigstore also supports signed timestamps. Trusted Timestamp Authorities (TSAs) issue signed timestamps following the RFC 3161 specification. |
+| **Transparency Log Verification** | Verifies inclusion of the signature in the **Rekor Transparency Log** for audibility. Not standalone verification. | Checked the Rekor Transparency Log to ensure signatures are publicly recorded. |
+
+### Cosign Keyless Verify Input
+
+1. The Artifact to Verify
+    - The OCI artifacts.
+
+2. Verification Materials
+    - Identity Parameters: Restrict verification to specific OIDC identities. Valid values include email address, DNS names, IP addresses, and URIs.
+    - OIDC Issuer Parameters: Ensure the certificate was issued by a specific OIDC provider, without Fulcio roots (for BYO PKI).
+    - Certificate Chain(Optional): Provide user-provided trusted chain for verification.
+    - Rekor URL(Optional): Specify a custom Rekor instance, default value is `https://rekor.sigstore.dev`.
+
+### Cosign Key-based Verify Input
+
+1. The Artifacts to Verify
+    - The OCI artifacts.
+
+2. Verification Materials
+    - Public Key: The public key in PEM format, corresponding to the private key used to sign the artifact.
+    - Certificate Chain(Optional): Provide user-provided trusted chain for verification.
+    - Rekor URL(Optional): Specify a custom Rekor instance, default value is https://rekor.sigstore.dev.
```

Comment:
curious if the rekor url will make a difference to the keyed verification

Reply:
Its like the customized `fulcio`, Sigstore provides Rekor as an implementation of the transparency log service, user can replace with their own service.